src/support.rs

```diff
@@ -106,7 +106,7 @@ impl<T> From<UniqueRef<T>> for UniquePtr<T> {
 pub struct UniqueRef<T: ?Sized>(NonNull<T>);
 
 impl<T> UniqueRef<T> {
-  unsafe fn try_from_raw(ptr: *mut T) -> Option<Self> {
+  pub unsafe fn try_from_raw(ptr: *mut T) -> Option<Self> {
```

Comment:
```suggestion
  pub(crate) unsafe fn try_from_raw(ptr: *mut T) -> Option<Self> {
```
Or is it intentional to promote it to the public API?

Reply:
I think `try_from_raw` and `from_raw` should have the same visibility since `from_raw` is just a thin wrapper that unwraps the result of `try_from_raw`?

But instead of promoting `try_from_raw` to the public API I think making both functions `pub(crate)` might make more sense, since there is special requirement on the pointer (`T`'s own `Drop` implementation should free the pointer) which doesn't seem explicit, and none of the current APIs expect externally-constructed `UniqueRef`.

Reply:
I agree, they're better off as `pub(crate)`.

Reply:
Fixed